File: explorer/src/main/java/com/duongame/task/LoadGifTask.java
package com.duongame.task;

import android.os.AsyncTask;

import java.io.DataInputStream;
import java.io.File;
import java.io.FileInputStream;
import java.io.FileNotFoundException;
import java.io.IOException;

/**
 * Created by namjungsoo on 2017. 1. 28..
 */

// AsyncTask:
//  parameter
//  progress
//  result
public class LoadGifTask extends AsyncTask<String, Integer, Void> {
    public interface LoadGifListener {
        public void onSuccess(byte[] data);

        public void onFail();
    }

    private byte[] data;
    final private LoadGifListener listener;

    public LoadGifTask(LoadGifListener listener) {
        this.listener = listener;
    }

    public static byte[] loadGif(String path) throws IOException {
        final File file = new File(path);
        byte[] data = new byte[(int) file.length()];
        final FileInputStream fis = new FileInputStream(file);
        final DataInputStream dis = new DataInputStream(fis);
        dis.readFully(data);
        dis.close();
        fis.close();
        return data;
    }

    @Override
    protected Void doInBackground(String... params) {
        final String path = params[0];

        try {
            final File file = new File(path);
            data = new byte[(int) file.length()];
            final FileInputStream fis = new FileInputStream(file);
            final DataInputStream dis = new DataInputStream(fis);
            dis.readFully(data);
            dis.close();
            fis.close();
        } catch (FileNotFoundException e) {
            e.printStackTrace();
            if (listener != null) {
                listener.onFail();
            }
        } catch (IOException e) {
            e.printStackTrace();
            if (listener != null) {
                listener.onFail();
            }
        }
        return null;
    }

    @Override
    protected void onPostExecute(Void result) {
        if (listener != null) {
            listener.onSuccess(data);
        }
    }
}

File: Explorer/src/main/java/com/duongame/task/bitmap/LoadGifTask.kt
package com.duongame.task.bitmap

import android.os.AsyncTask
import java.io.*

/**
 * Created by namjungsoo on 2017. 1. 28..
 */
// AsyncTask:
//  parameter
//  progress
//  result
class LoadGifTask(private val listener: LoadGifListener?) : AsyncTask<String?, Int?, Void?>() {
    interface LoadGifListener {
        fun onSuccess(data: ByteArray?)
        fun onFail()
    }

    private lateinit var data: ByteArray
    protected override fun doInBackground(vararg params: String?): Void? {
        val path = params[0]
        try {
            val file = File(path)
            data = ByteArray(file.length().toInt())
            val fis = FileInputStream(file)
            val dis = DataInputStream(fis)
            dis.readFully(data)
            dis.close()
            fis.close()
        } catch (e: FileNotFoundException) {
            e.printStackTrace()
            listener?.onFail()
        } catch (e: IOException) {
            e.printStackTrace()
            listener?.onFail()
        }
        return null
    }

    override fun onPostExecute(result: Void?) {
        listener?.onSuccess(data)
    }

    companion object {
        @Throws(IOException::class)
        fun loadGif(path: String?): ByteArray {
            val file = File(path)
            val data = ByteArray(file.length().toInt())
            val fis = FileInputStream(file)
            val dis = DataInputStream(fis)
            dis.readFully(data)
            dis.close()
            fis.close()
            return data
        }
    }
}